### PR TITLE
Fix issue with drain=False in the adaptive optimizer

### PR DESCRIPTION
**Pull request by soranjh**
**Context:**
The adaptive optimizer requires the operator pool to be re-constructed inside the optimization pool when `drain_pool=False`. This PR fixes this by copying the added gate inside the `append_gate` function.

**Description of the Change:**

- The following line is added to the `append_gate` function: `g = g.__copy__()`
- `test_step_and_cost_nodrain` test is modified to prevent having the same issue in the future.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/optimize/adaptive.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ def append_gate(tape, params, gates):
         qml.apply(o)
 
     for i, g in enumerate(gates):
+        g = g.__copy__()
```
Comment (soranjh) [inline pennylane/optimize/adaptive.py:34]: `g = copy.copy(g)` also works here. Not sure if it has any other drawbacks.

Comment (josh146) [reply pennylane/optimize/adaptive.py:34]: @soranjh I think `copy` is actually preferred here!

File: pennylane/optimize/adaptive.py [hunk 2]
```diff
@@ -204,9 +205,7 @@ def step_and_cost(self, circuit, operator_pool, drain_pool=False, params_zero=Tr
         if params_zero:
             params = np.zeros(len(selected_gates))
         else:
-            params = np.array(
-                [gate.parameters[0]._value for gate in selected_gates], requires_grad=True
-            )
+            params = np.array([gate.parameters[0] for gate in selected_gates], requires_grad=True)
```
Comment (josh146) [inline pennylane/optimize/adaptive.py:208]: How come this was changed?

**Review by josh146 — APPROVED** [review]
Thanks @soranjh, don't forget to update the changelog!

File: pennylane/optimize/adaptive.py (hunk 2 above)
Comment (soranjh) [reply pennylane/optimize/adaptive.py:208]: It was failing a test after adding `g = g.__copy__()` with:

`FAILED test_adaptive.py::test_qubit_rotation[qubit_rotation_circuit] - AttributeError: 'tensor' object has no attribute '_value'`

Before copying the gate, the type of the added gate parameter was: `autograd.numpy.numpy_boxes.ArrayBox`. Now that we copy the gate, the type of the parameter is `pennylane.numpy.tensor.tensor` and `._value` is not needed anymore.

**Pull request merged**